core/org.wso2.carbon.ui/src/main/java/org/wso2/carbon/ui/CarbonUILoginUtil.java

```diff
@@ -120,7 +120,14 @@ protected static boolean saveOriginalUrl(CarbonUIAuthenticator authenticator,
                 cookie.setPath("/");
                 cookie.setSecure(true);
                 cookie.setHttpOnly(true);
-                response.addCookie(cookie);
+
+                try {
+                    response.addCookie(cookie);
+                } catch (IllegalArgumentException e) {
+                    if (log.isDebugEnabled()) {
```

Comment:
We can remove the `log.isDebugEnabled()` check as the log entry does not contain any string operations.

Reply:
Addressed in https://github.com/wso2/carbon-kernel/pull/3551

Reply:
Just out of curiosity, why we are using debug log for a caught exception which is not thrown?

Reply:
Please refer to https://github.com/wso2/carbon-kernel/pull/3339/commits/b18b61cc3f9309208a30ea535a21553de6164362#r909283272
Since this is a client error and does not have severe impact on the rest of the flow, we decided to go with a debug log.